mamar-web/src/app/store/variation.ts

```diff
@@ -73,6 +81,32 @@ export function variationReducer(variation: Variation, action: VariationAction):
         } else {
             return variation
         }
+    case "add_segment":
+        const newSeg: Segment = {
+            type: "Subseg",
+            id: action.id,
+            trackList: action.trackList,
+        }
+        return {
+            ...variation,
+            segments: [
+                ...variation.segments,
+                newSeg,
+            ],
+        }
+    case "add_loop_start":
+        const startLoopSeg: Segment = {
+            id: action.id,
+            type: "StartLoop",
+            label_index: action.labelIndex,
+        }
+        return {
+            ...variation,
+            segments: [
+                ...variation.segments,
+                startLoopSeg,
```

Comment:
Should probably swap these two lines. StartLoop needs to go at the start. Also should add EndLoop, as having a StartLoop without an EndLoop after it is an error

Reply:
StartLoop shouldn't go at the start of the entire song, right? That seems like it'd go against the point of this PR

Reply:
It should be configurable yeah. But certainly not at the end, there should be a segment between it

Perhaps add
StartLoop
empty segment
EndLoop

Reply:
Sounds good, provided you can then import a midi to be used by that empty segment